lib/src/parse/stylesheet.dart

```diff
@@ -1753,30 +1766,52 @@ abstract class StylesheetParser extends Parser {
 
   /// Consumes the contents of a hex color, after the `#`.
   SassColor _hexColorContents(LineScannerState start) {
-    var red = _hexDigit();
-    var green = _hexDigit();
-    var blue = _hexDigit();
-
-    var next = scanner.peekChar();
-    if (next != null && isHex(next)) {
-      red = (red << 4) + green;
-      green = (blue << 4) + _hexDigit();
-      blue = (_hexDigit() << 4) + _hexDigit();
+    var digit1 = _hexDigit();
```

Comment:
Like this change to split the variables into digit1/2/3 initially and then red/green/blue once we figure out what a digit means. 👍